lib/active_force/active_query.rb

```diff
@@ -118,6 +120,83 @@ def order *args
 
     private
 
+    def build_includes(relation)
+      association = sobject.associations[relation]
+      fields Association::EagerLoadProjectionBuilder.build(association)
+      association_mapping[association.sfdc_association_field.downcase] = association.relation_name
+    end
+
+    def build_hash_includes(relation)
+      relation.each do |key, value|
+        association = sobject.associations[key]
+        association_name = association.class.name.split('::').last
+        case association_name
+        when 'HasManyAssociation', 'HasOneAssociation'
+          fields build_relation(association, value)
+        when 'BelongsToAssociation'
+          fields build_relation_for_belongs_to(association.sfdc_association_field, association, value)
+        end
```

Comment:
```suggestion
        association = sobject.associations[key]
        case association
        when ActiveForce::Association::BelongsToAssociation
          fields build_relation_for_belongs_to(association.sfdc_association_field, association, value)
        else
          fields build_relation(association, value)
        end
```
Can you simplify this entire block in this way?  You may need to `require` the `belongs_to_association.rb` in order for the constant lookup to work.

Reply:
updated on latest commits